monix-tail/shared/src/main/scala/monix/tail/internal/IterantFoldWhileLeft.scala

```diff
@@ -45,26 +47,30 @@ private[tail] object IterantFoldWhileLeft {
       if (hasResult)
         stop.map(_ => s)
       else
-        rest.flatMap(loop(s))
+        rest.flatMap(loop(stopRef)(s))
     }
 
-    def loop(state: S)(self: Iterant[F, A]): F[S] = {
+    def loop(stopRef: ObjectRef[F[Unit]])(state: S)(self: Iterant[F, A]): F[S] = {
```

Comment:
We don't need 3 parameter groups for this curried function. `stopRef` and `state` can be on the same level so to speak.